test/unit/services/query-service.test.js

```diff
@@ -209,5 +340,123 @@ describe('query-service', () => {
         expect.objectContaining({ prompt })
       )
     })
+
+    test('fetchAnswer does not return summary answer when summariesEnabled is false', async () => {
+      createRetrievalChain.mockResolvedValue({
+        invoke: jest.fn().mockResolvedValue({
+          answer: JSON.stringify({ answer: 'full index response', items: [] }),
+          context: [
+            {
+              pageContent:
+                '(Title: True title | Grant Scheme Name: True scheme | Source: https://www.gov.uk/link | Chunk Number: 0)===True summary',
+              metadata: {}
+            }
+          ]
+        })
+      })
+
+      const prompt = jest.fn()
+      const input = 'deer fencing'
+      ChatPromptTemplate.fromMessages.mockReturnValue(prompt)
+
+      const { response, summariesMode } = await fetchAnswer({}, input, [], false, false)
+
+      expect(JSON.parse(response).answer).toStrictEqual(JSON.stringify({ answer: 'full index response', items: [] }))
+      expect(createStuffDocumentsChain).toHaveBeenCalledWith(
+        expect.objectContaining({ prompt })
+      )
+      expect(summariesMode).toBe(false)
+    })
+
+    test('fetchAnswer does not fall back to full index when summary answer is valid', async () => {
+      createRetrievalChain.mockResolvedValueOnce({
+        invoke: jest.fn().mockResolvedValue({
+          answer: JSON.stringify({ answer: 'summary response', items: [] }),
+          context: [
+            {
+              pageContent:
+                '(Title: True title | Grant Scheme Name: True scheme | Source: https://www.gov.uk/link | Chunk Number: 0)===True summary',
+              metadata: {}
+            }
+          ]
+        })
+      })
+
+      const prompt = jest.fn()
+      const input = 'deer fencing'
+      ChatPromptTemplate.fromMessages.mockReturnValue(prompt)
+
+      const { response } = await fetchAnswer({}, input, [], false, true)
+
+      expect(JSON.parse(response).answer).toStrictEqual(JSON.stringify({ answer: 'summary response', items: [] }))
+      expect(createStuffDocumentsChain).toHaveBeenCalledWith(
+        expect.objectContaining({ prompt })
+      )
+    })
+  })
+
+  describe('validateResponseLinks', () => {
```

Comment:
Merge mistake? These two `validateResponseLinks` tests have been moved to test/unit/utils/validators.test.js